js/modules/loki_message_api.js

```diff
@@ -270,29 +165,30 @@ class LokiMessageAPI {
     const doRequest = async (nodeUrl, nodeData) => {
       const params = {
         pubKey: ourKey,
-        lastHash: nodeData.lastHash,
+        lastHash: nodeData.lastHash || '',
       };
 
       try {
         const result = await rpc(
-          nodeUrl,
-          this.messageServerPort,
+          `http://${nodeUrl}`,
+          this.snodeServerPort,
           'retrieve',
           params
         );
 
         nodeComplete(nodeUrl);
 
-        if (result.lastHash) {
-          lokiSnodeAPI.updateLastHash(nodeUrl, result.lastHash);
+        if (Array.isArray(result.messages) && result.messages.length) {
+          const lastHash = [...result.messages].pop().hash;
```

Comment:
I remember the reason why we had the server return the last hash was that we didn't want our system to rely on the order of the messages, especially when we have multiple snodes returning slightly different subsets of the messages due to the intra-swarm propagation. The risk was that the client would save his last hash and basically miss messages that are considered older than that last hash... It's a complex issue. Do we have more reassurance now it's going to work?

Reply:
Ya because the client stores a different last hash for each snode it shouldn't matter if the snodes store the messages in different orders, and we always build the json in the same order that the messages are stored in the storage server database. So the lastHash value has actually always been the hash of the last message in the list I'm pretty sure

Reply:
Cool. Just needed to be reminded :)